src/main/java/com/breskul/hw/controllers/NasaController.java

```diff
@@ -0,0 +1,78 @@
+package com.breskul.hw.controllers;
+
+import com.breskul.hw.controllers.entity.Photos;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.springframework.http.HttpStatus.MOVED_PERMANENTLY;
+
+@Component
+public class NasaController {
+
+    Logger logger = LoggerFactory.getLogger(NasaController.class);
+
+    final String url = "https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos";
+    RestTemplate restTemplate = new RestTemplate();
+
+    public byte[] getLargestPicture(int sol, Optional<String> camera) {
```

Comment:
It is a bad practice to use `Optional` as a method argument. It would be better to pass the value and wrap it to Optional.ofNullable() inside the method in case you need a null-check